nexus/src/app/sagas/instance_create.rs

```diff
@@ -774,17 +865,20 @@ async fn sic_create_instance_record(
 }
 
 async fn sic_delete_instance_record(
-    sagactx: ActionContext<SagaInstanceCreate>,
+    sagactx: NexusActionContext,
 ) -> Result<(), anyhow::Error> {
     let osagactx = sagactx.user_data();
-    let params = sagactx.saga_params();
+    let params = sagactx.saga_params::<Params>()?;
     let datastore = osagactx.datastore();
     let opctx = OpContext::for_saga_action(&sagactx, &params.serialized_authn);
     let instance_id = sagactx.lookup::<Uuid>("instance_id")?;
     let instance_name = sagactx.lookup::<db::model::Name>("instance_name")?;
 
     // We currently only support deleting an instance if it is stopped or
     // failed, so update the state accordingly to allow deletion.
+    // TODO-correctness TODO-security It's not correct to re-resolve the
```

Comment:
Issue?

Reply:
Filed https://github.com/oxidecomputer/omicron/issues/1536 and updated this comment to reference it.